### ISSUE TITLE
make relay unit more robust

### ISSUE THREAD

**Issue by csuzhangxc** [pingcap/dm#91]
## Feature Request

**Is your feature request related to a problem? Please describe:**

Currently, we need the following assumptions to make relay unit workers correctly:

1. master always sends all binlog events in its binlog file
2. master sends an event until all of the previous events sent

But these assumptions may not always satisfied.

**Describe the feature you'd like:**

We can change the assumption to the following:

- **master always sends all binlog events a slave needed**

In other words, the correctness of the relay unit relies on **the sending behavior of the master**, not binlog file on the master.

In practice, we can use some dummy events to fill the relay log file if no corresponding events received from the master.
